### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by wraithgar**
This is a rewrite of proc-log

- **feat!: change current exported function "log"**
    Everything that's currently in proc-log is now exported as a `log` object
- **fix(refactor): don't loop through levels to generate log export**
    The entire module is inline, no loops.  Tests ensure that we are synced between levels and methods.
- **feat(log): add timing log level**
    A new `timing` log level has been added
- **feat: add output methods**
    A new `output` export has been added with three levels: `standard`, `error`, and `buffer`. It works exactly like the `log` object but emits `output` events.
- **fix: switch to traditional functions**
    They are marginally faster and this is potentially a pretty hot path.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,47 +1,56 @@
...
 
+```js
+const { log, output } = require('proc-log')
+```
+
+#### output
+* `output.standard(...args)` calls `process.emit('output', 'standard', ...args)`
+  This is for general standard output.  Consumers will typically show this on stdout (after optionally formatting or filtering it).
+* `output.error(...args)` calls `process.emit('output', 'error', ...args)`
+  This is for general standard output.  Consumers will typically show this on stderr (after optionally formatting or filtering it).
```
Comment (lukekarrys) [inline README.md:19]:
```suggestion
  This is for general error output.  Consumers will typically show this on stderr (after optionally formatting or filtering it).
```

File: test/index.js [hunk 2]
```diff
@@ -1,16 +1,32 @@
...
-
-t.matchSnapshot(LEVELS, 'log levels')
-for (const level of LEVELS) {
-  t.test(level, t => {
-    t.match(log[level], Function)
-    process.once('log', (actual, ...args) => {
-      t.equal(actual, level, 'emitted log with expected level')
-      t.same(args, [1, 2, 3], 'got expected args')
+const procLog = require('../')
+for (method in procLog) {
```
Comment (lukekarrys) [inline test/index.js:3]:
```suggestion
for (const method in procLog) {
```

Comment (wraithgar) [reply test/index.js:3]: I promise I'm a professional programmer.

File: test/index.js [hunk 3]
```diff
@@ -1,16 +1,32 @@
...
-  t.test(level, t => {
-    t.match(log[level], Function)
-    process.once('log', (actual, ...args) => {
-      t.equal(actual, level, 'emitted log with expected level')
-      t.same(args, [1, 2, 3], 'got expected args')
+const procLog = require('../')
+for (const method in procLog) {
+  t.test(method, t => {
+    const log = procLog[method]
+    const { LEVELS } = log
```
Comment (fritzy) [inline test/index.js:6]: This tests that the methods and log levels match, but it feels a bit fragile. We're testing that things are internally consistent, but if there's a bug in this test, we wouldn't know that it's not testing the breadth of level/functions we expect. Even a simple `LEVELS.length >= 10` assertion would prove that we're looping through and testing _something_.

**Review by fritzy — CHANGES_REQUESTED** [review]

File: test/index.js (hunk 3 above)
Comment (wraithgar) [reply test/index.js:6]: We snapshot the levels to make sure there's something there

```
t.matchSnapshot(LEVELS, `${method} levels`)
```

Comment (wraithgar) [reply test/index.js:6]: a `t.plan` could make sure we're testing the expected two top level methods?

Comment (wraithgar) [reply test/index.js:6]:
```js
const procLog = require('../')
// This makes sure we are testing the two known exported methods.
t.plan(2)
for (const method in procLog) {
```

Comment (wraithgar) [reply test/index.js:6]: Or we can snapshot `Object.keys(procLog)`?

Comment (fritzy) [reply test/index.js:6]: I didn't notice the snapshot of `LEVELS`. Approving...

**Review by fritzy — DISMISSED** [review]
👍

File: test/index.js (hunk 3 above)
Comment (wraithgar) [reply test/index.js:6]: tap snapshots are weird in that if I removed a level, that snapshot would live in the snapshot file with no errors. I am adding the `t.plan(2)` we we at least are asserting that we have two main exports before we iterate throught them.

**Review by fritzy — APPROVED** [review]

**Review by lukekarrys — APPROVED** [review]

**Pull request merged**